python/lancedb/query.py

```diff
@@ -572,3 +606,188 @@ def to_arrow(self) -> pa.Table:
+        )
+        return results
+
+    def _normalize_scores(self, results: pa.Table, invert=False):
+        if len(results) == 0:
+            return results
+        # Get the _score column from results
+        scores = results.column("_score").to_numpy()
+        # normalize the scores by subtracting the min and dividing by the max
+        max, min = np.max(scores), np.min(scores)
+        if np.isclose(max, min):
+            rng = max
+        else:
+            rng = max - min
+        scores = (scores - min) / rng
+        if invert:
+            scores = 1 - scores
+        # replace the _score column with the ranks
+        _score_idx = results.column_names.index("_score")
+        results = results.set_column(
+            _score_idx, "_score", pa.array(scores, type=pa.float32())
+        )
+        return results
+
+    def _merge_results(
+        self, vector_results: pa.Table, fts_results: pa.Table, fill: float
+    ):
+        # If both are empty then just return an empty table
+        if len(vector_results) == 0 and len(fts_results) == 0:
+            return vector_results
+        # If one is empty then return the other
+        if len(vector_results) == 0:
+            return fts_results
+        if len(fts_results) == 0:
+            return vector_results
+
+        # sort both input tables on _rowid
+        combined_list = []
+        vector_list = vector_results.sort_by("_rowid").to_pylist()
+        fts_list = fts_results.sort_by("_rowid").to_pylist()
+        i, j = 0, 0
+        while i < len(vector_list):
+            if j >= len(fts_list):
+                for vi in vector_list[i:]:
+                    vi["_score"] = self._combine_score(vi["_rowid"], fill)
+                    combined_list.append(vi)
+                break
+
+            vi = vector_list[i]
+            fj = fts_list[j]
+            if vi["_rowid"] == fj["_rowid"]:
+                vi["_score"] = self._combine_score(vi["_rowid"], fj["_rowid"])
+                combined_list.append(vi)
+                i += 1
+                j += 1
+            elif vector_list[i]["_rowid"] < fts_list[j]["_rowid"]:
+                vi["_score"] = self._combine_score(vi["_rowid"], fill)
+                combined_list.append(vi)
+                i += 1
+            else:
+                fj["_score"] = self._combine_score(fj["_rowid"], fill)
+                combined_list.append(fj)
+                j += 1
+        if j < len(fts_list) - 1:
+            for fj in fts_list[j:]:
+                fj["_score"] = self._combine_score(fj["_rowid"], fill)
+                combined_list.append(fj)
+        return pa.Table.from_pylist(combined_list, schema=vector_results.schema)
+
+    def _combine_score(self, score1, score2):
+        return self._weight * score1 + (1 - self._weight) * score2
+
+    def rerank(self, weight=0.7, normalize="score", reranker:Union[str, callable] = None) -> LanceHybridQueryBuilder:
```

Comment:
`Option[Union[str, callable]]`

Reply:
please add some code docs to describe the behavior. My assumption is:
1. If reranker is `str` it's assumed to be one of the known rerankers
2. If reranker is `callable` it's a custom function? what's the function api need to look like?
3. If None then use weight to do a linear combination?

If a reranker is specified, does normalize still apply?

Reply:
Ohh just need to update the type annotation--was using this before Reranker base class